src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/FrameworkEventSource.cs

```diff
@@ -94,25 +95,27 @@ public void ThreadPoolEnqueueWork(long workID)
             WriteEvent(30, workID);
         }
 
+        // The object's current location in memory was being used before. Since objects can be moved, it may be difficult to
+        // associate Enqueue/Dequeue events with the object's at-the-time location in memory, the ETW listeners would have to
+        // know specifics about the events and track GC movements to associate events. The hash code is a stable value and
+        // easier to use for association, though there may be collisions.
         [NonEvent]
-        public unsafe void ThreadPoolEnqueueWorkObject(object workID)
-        {
-            // convert the Object Id to a long
-            ThreadPoolEnqueueWork((long)*((void**)Unsafe.AsPointer(ref workID)));
-        }
+        [MethodImpl(MethodImplOptions.NoInlining)]
```

Comment:
Why do these methods need to be marked as non-inlineable?

Reply:
These are called from hot paths and inlining them makes the caller's code larger and the hot path more sparse

Reply:
Gotcha.  Makes sense.